src/tests/ftest/tags.py

```diff
@@ -23,6 +24,22 @@
 STAGE_FREQUENCY_TAGS = ('all', 'pr', 'daily_regression', 'full_regression')
 
 
+class TagSet(set):
+    """Set with handling for negative entries."""
+
+    def issubset(self, other):
+        for tag in self:
+            if tag.startswith('-'):
+                if tag[1:] in other:
+                    return False
+            elif tag not in other:
+                return False
+        return True
+
+    def issuperset(self, other):
+        return TagSet(other).issubset(self)
```

Comment:
Maybe this is technically faster
```suggestion
    def issuperset(self, other):
        return TagSet.issubset(other, self)
```